jib-core/src/main/java/com/google/cloud/tools/jib/ncache/CacheWriteEntry.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2018 Google LLC. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.ncache;
+
+import com.google.cloud.tools.jib.blob.Blob;
+import com.google.cloud.tools.jib.image.DescriptorDigest;
+import java.util.Optional;
+
+/** Represents layer data to write to the cache. <b>Implementations must be immutable.</b> */
+public interface CacheWriteEntry {
```

Comment:
Having a write entry feels weird.  How would I create an instance of this?

Reply:
Via static factories that builds the `CacheWriteEntry` for specific use cases (https://github.com/GoogleContainerTools/jib/pull/874/files#diff-d25f928c70b6196718d3234abbdbf5e3R33)